presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -139,12 +139,24 @@ public StatisticRange intersect(StatisticRange other)
         return empty();
     }
 
-    public StatisticRange add(StatisticRange other)
+    public StatisticRange addAndSumDistinctValues(StatisticRange other)
     {
         double newDistinctValues = distinctValues + other.distinctValues;
         return new StatisticRange(minExcludeNaN(low, other.low), maxExcludeNaN(high, other.high), newDistinctValues);
     }
 
+    public StatisticRange addAndCollapseDistinctValues(StatisticRange other)
+    {
+        double overlapPercentOfLeft = this.overlapPercentWith(other);
```

Comment:
use `This` and `Other` instead `Left` and `Right` as suffix for variables to be consistent with call signature.